Add the ability to request and receive blocks to the Agora node

MichaelKim20 opened the pull request:
This will only be used for restoration purposes.

Libraries 'axios' and 'urijs' are changed from development
to production for data requests to Agora nodes

Related to #45

TrustHenry commented on src/modules/agora/AgoraClient.ts:
```diff
@@ -0,0 +1,131 @@
+     */
+    constructor (host: string, port: string)
+    {
+        this.client = axios.create();
```
```
this.client = axios.create({
            baseURL: `url:port`
        });
```

Don't it need baseURL basic settings for `Axios.create`?

TrustHenry replied:
`this.client.defaults.timeout = 3000;`

It can also set a timeout.

MichaelKim20 replied:
Thank you for your good suggestion.
I changed it as below.

```
    /**
     * The URL encoder
     */
    private uri: URI;

    /**
     * Constructor
     * @param host - The network address to connect to the Agora
     * @param port - The network port to connect to the Agora
     */
    constructor (host: string, port: string)
    {
        this.client = axios.create();
        this.client.defaults.timeout = 10000;
        this.uri = URI(host).port(port);
    }
```

TrustHenry reviewed (approved):
LGTM

Geod24 commented on src/modules/agora/AgoraClient.ts:
```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+
+    The class that recover data
```
```suggestion
    The class that recovers data
```

MichaelKim20 replied:
I changed it

Geod24 replied:
It's still here ?

MichaelKim20 replied:
Yeah, I revised the 18th line.

Geod24 commented on src/modules/agora/AgoraClient.ts:
```diff
@@ -0,0 +1,125 @@
+
+    /**
+     * Constructor
+     * @param host - The network address to connect to the Agora
```
You can either write "connect to Agora" or "connect to the Agora node", but not "connect to the Agora".
Likewise, you can say "going to Samsung" or "going to the Samsung headquarter", not "going to the Samsung".

MichaelKim20 replied:
I changed it

Geod24 commented on src/modules/agora/AgoraClient.ts:
```diff
@@ -0,0 +1,125 @@
+    {
+        return new Promise<Array<any>>((resolve, reject) =>
+        {
+            this.uri
+                .directory("blocks_from")
+                .setSearch("block_height", block_height.toString())
+                .setSearch("max_blocks", max_blocks.toString());
```
If I understand correctly, this modifies the `uri` member ? Doesn't that mean that this function is not re-entrant, which is at odds with its `Promise`-based approach ?

MichaelKim20 replied:
The address and port of the client need not be preset. Because I have to enter the entire URL every time I call this.client.get.
So as an alternative, I set the address and port on the URL.

I checked in advance through the test.
I have confirmed that it is reset.
```
describe ('Test of Stoa API Server', () => {
    let host: string = 'http://localhost';
    let port: string = '3837';

    it('Test of URI', () => {
        let uri = URI(host)
            .port(port);

        uri
            .directory("1")
            .setSearch("block_height", "1")
            .setSearch("max_blocks", "1");
        assert.strictEqual(uri.toString(), "http://localhost:3837/1/?block_height=1&max_blocks=1");

        uri
            .directory("2")
            .setSearch("block_height", "2")
            .setSearch("max_blocks", "2");
        assert.strictEqual(uri.toString(), "http://localhost:3837/2/?block_height=2&max_blocks=2");
    });
});
```

MichaelKim20 replied:
@Geod24 I'll check and ask for a review because there's something I need to revise.

MichaelKim20 replied:
@Geod24 That's right. The value of this.uri can be changed by other repeated calls.
So I changed it back to a local variable
Ready of review

Geod24 reviewed (changes requested):
Questions about the URI approach

MichaelKim20 committed:
Add the ability to request and receive blocks to the Agora

This will only be used for restoration purposes.
Libraries 'axios' and 'urijs' are changed from development
to production for data requests to Agora nodes
